app/controllers/application_controller.rb

```diff
@@ -1,8 +1,16 @@
 class ApplicationController < ActionController::Base
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
-  # Probably should move this after implementing errorhandler
+  rescue_from ChangeStatusError, with: :change_status_error
+
+  private
+
   def user_not_authorized
     flash[:alert] = "You are not authorized to perform this action."
     redirect_back(fallback_location: root_path)
   end
+
+  def change_status_error
```

Comment:
Why would you like to keep it here (`ApplicationController`, not `ReservationsController`?) It looks like this code it's strictly related to `reservation`, isn't it? :thinking: